### PR TITLE
Port of `Align cursor` extension from vscode

### PR DESCRIPTION
**Pull request by sirynka**
Inspired by: [align cursor](https://marketplace.visualstudio.com/items?itemName=yo1dog.cursor-align)

Ideally, I would have wanted to implement it as an actual extensions, instead of committing it into the core but have not figured how to do it yet. I'd be glad if you'd provide an example of how to interact with editor through the extension API.

> Nothing exists on that front, alas.

Progress:
- [x] Support multiple cursors in the same line
- [ ] Add tests for backwards selection
- [ ] Add the ability to align with tabs

Release Notes:

- Added `align cursor` action to `command pallete`

### CODE REVIEW

**Review by SomeoneToIgnore — CHANGES_REQUESTED** [review]
This needs tests.

**Comment by SomeoneToIgnore** [comment]
> I'd be glad if you'd provide an example of how to interact with editor through the extension API.

Nothing exists on that front, alas.

**Comment by sirynka** [comment]
> This needs tests.

Is this enough: 849ffca8f2082ba73b6008c7b1658b8a1cbf0b99?

**Comment by SomeoneToIgnore** [comment]
* to start with, read to the bot messages already

* Not sure how realistic my example is, but multiple ˇ on the same line seems like a good thing to check one way or another:

```py
while True:
        try:
ˇn = iˇnt(input("Enter the number of elements in the Fibonacci sequence: "))
    ˇif n <= 0:
                print("Please enter a positive integer greater than 0.")
                continue
            ˇbreak  # Exit the loop when a valid number is providedˇ
        except ValueError:
            print("Invalid input. Please enter an integer.")
```

File: crates/editor/src/editor.rs [hunk 1]
```diff
@@ -6378,6 +6378,27 @@ impl Editor {
         self.reload(project, cx).detach_and_notify_err(cx);
     }
 
+    pub fn align_cursor(&mut self, _: &AlignCursor, cx: &mut ViewContext<Self>) {
+        self.transact(cx, |this, cx| {
+            let max_len = this
+                .selections
+                .all::<Point>(cx)
+                .iter()
+                .map(|s| s.start.column)
```
Comment (SomeoneToIgnore) [inline crates/editor/src/editor.rs:6387]: What if we select backwards? Then `start` will be after the `end`.

File: crates/editor/src/editor.rs [hunk 2]
```diff
@@ -6378,6 +6378,27 @@ impl Editor {
...
+                .max()
+                .unwrap_or(0);
+
+            let mut changes = Vec::new();
+            for s in this.selections.all::<Point>(cx).iter() {
+                let spaces = " ".repeat((max_len - s.start.column) as usize);
```
Comment (SomeoneToIgnore) [inline crates/editor/src/editor.rs:6393]: The user may use tabs, is `" "` a good choice always?

Comment (SomeoneToIgnore) [reply crates/editor/src/editor.rs:6393]: Also same question as above, 

> What if we select backwards? Then `start` will be after the `end`.


In addition to that, does this all make sense when `max_len == 0`? 
Looks like a noop, and `0 - something` may actually panic, which is super bad.
This edge case needs a test.

**Review by SomeoneToIgnore — CHANGES_REQUESTED** [review]

File: crates/editor/src/editor.rs [hunk 3]
```diff
@@ -6378,6 +6378,72 @@ impl Editor {
...
+                }
+                b
+            };
+
+            let colwise_offsets = |a: Vec<Vec<u32>>| -> Vec<Vec<u32>> {
+                let a = transpose(a);
+                let a = offsets(a);
+                let a = transpose(a);
+                a
+            };
```
Comment (sirynka) [inline crates/editor/src/editor.rs:6432]: It could definitely be written in a smarter way.

**Comment by sirynka** [comment]
> - to start with, read to the bot's messages already

I don't understand what else does `cla-bot` wants from me.
I have configured `user.email` to the same one that is set in my github profile
And unchecked **Keep my email addresses private** checkbox in profile settings.

**Comment by SomeoneToIgnore** [comment]
The commits in this PR seem to be not updated: https://github.com/zed-industries/zed/pull/20129/commits
Have you updated the old commits' metadata?

**Comment by sirynka** [comment]
@cla-bot check
